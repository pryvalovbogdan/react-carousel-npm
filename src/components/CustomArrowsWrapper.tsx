import React from 'react';

import { renderChild } from '../utils';
import HeaderArrows from './HeaderArrows';
import { ICustomArrowsWrapper } from '../types/CarouselTypes';

const CustomArrowsWrapper = ({
  CustomArrowBtn,
  disabled,
  currentPage,
  lastPage,
  handlePrevPage,
  handleNextPage,
}: ICustomArrowsWrapper) => {
  return (
    <div>
      {CustomArrowBtn ? (
        <>
          {renderChild(CustomArrowBtn, {
            disabled: disabled || currentPage === 1,
            onClick: handlePrevPage,
            isLeftArrow: true,
          })}
          {renderChild(CustomArrowBtn, {
            disabled: disabled || currentPage === lastPage,
            onClick: handleNextPage,
            isLeftArrow: false,
          })}
        </>
      ) : (
        <HeaderArrows
          disabled={disabled}
          lastPage={lastPage}
          currentPage={currentPage}
          goToNextPage={handleNextPage}
          goToPrevPage={handlePrevPage}
        />
      )}
    </div>
  );
};

export default CustomArrowsWrapper;
