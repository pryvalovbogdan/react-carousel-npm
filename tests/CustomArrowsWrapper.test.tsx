import { render } from '@testing-library/react';

import React from 'react';
import CustomArrowsWrapper from '../src/components/CustomArrowsWrapper';

describe('Test CustomArrowsWrapper', () => {
  const CustomArrowBtn = <div>Child content</div>;

  test('snapshot Notes', () => {
    const { baseElement } = render(
      <CustomArrowsWrapper
        currentPage={1}
        CustomArrowBtn={CustomArrowBtn}
        disabled={false}
        lastPage={2}
        handlePrevPage={() => {}}
        handleNextPage={() => {}}
      />,
    );

    expect(baseElement).toMatchSnapshot();
  });
});
