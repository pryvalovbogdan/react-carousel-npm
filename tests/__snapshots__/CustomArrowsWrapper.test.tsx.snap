// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test CustomArrowsWrapper snapshot Notes 1`] = `
<body>
  <div>
    <div>
      <div
        disabled=""
      >
        Child content
      </div>
      <div>
        Child content
      </div>
    </div>
  </div>
</body>
`;
